src/main/java/com/uid2/admin/vertx/service/CloudEncryptionKeyService.java

```diff
@@ -3,39 +3,85 @@
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.uid2.admin.auth.AdminAuthMiddleware;
+import com.uid2.admin.cloudEncryption.CloudKeyRotationStrategy;
 import com.uid2.admin.model.CloudEncryptionKeyListResponse;
 import com.uid2.admin.model.CloudEncryptionKeySummary;
+import com.uid2.admin.store.writer.CloudEncryptionKeyStoreWriter;
 import com.uid2.admin.vertx.Endpoints;
 import com.uid2.shared.auth.Role;
+import com.uid2.shared.model.CloudEncryptionKey;
 import com.uid2.shared.store.reader.RotatingCloudEncryptionKeyProvider;
+import com.uid2.shared.store.reader.RotatingSiteStore;
 import com.uid2.shared.util.Mapper;
 import io.vertx.core.http.HttpHeaders;
 import io.vertx.ext.web.Router;
 import io.vertx.ext.web.RoutingContext;
 
-import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
 import java.util.stream.Collectors;
-import java.util.stream.Stream;
 
 public class CloudEncryptionKeyService implements IService {
     private final AdminAuthMiddleware auth;
     private final RotatingCloudEncryptionKeyProvider keyProvider;
+    private final RotatingSiteStore siteProvider;
+    private final CloudEncryptionKeyStoreWriter keyWriter;
+    private final CloudKeyRotationStrategy rotationStrategy;
     private static final ObjectMapper OBJECT_MAPPER = Mapper.getInstance();
 
-    public CloudEncryptionKeyService(AdminAuthMiddleware auth, RotatingCloudEncryptionKeyProvider keyProvider) {
+    public CloudEncryptionKeyService(
+            AdminAuthMiddleware auth,
+            RotatingCloudEncryptionKeyProvider keyProvider,
+            CloudEncryptionKeyStoreWriter keyWriter,
+            RotatingSiteStore siteProvider,
+            CloudKeyRotationStrategy rotationStrategy
+    ) {
         this.auth = auth;
         this.keyProvider = keyProvider;
+        this.siteProvider = siteProvider;
+        this.keyWriter = keyWriter;
+        this.rotationStrategy = rotationStrategy;
     }
 
     @Override
     public void setupRoutes(Router router) {
         router.get(Endpoints.CLOUD_ENCRYPTION_KEY_LIST.toString()).handler(
                 auth.handle(this::handleList, Role.MAINTAINER)
         );
+
+        router.post(Endpoints.CLOUD_ENCRYPTION_KEY_ROTATE.toString()).handler(
+                auth.handle(this::handleRotate, Role.MAINTAINER)
```

Comment:
Will this role allow for cron job to call it?

Reply:
Ah good point. We actually don't have a role for cloud encryption stuff - encryption job is using PRIVATE_OPERATOR_SYNC. Both salt and encryption key rotation use `SECRET_ROTATION`, that may be reasonable to use here as well.